### PR TITLE
Add a flag openstack_post_provisioning_script

### PR DESCRIPTION
**Pull request by keerockl**

### CODE REVIEW

**Comment by keerockl** [comment]
Could you please kindly let me know which metadata file you are meaning?
I have already specified default value as below flags.DEFINE_string('openstack_post_provisioning_script', None, '...

**Comment by jellyfishcake** [comment]
Ah, what I mean is you want to add a method in OpenStackVirtualMachine class that adds onto GetResourceMetadata in virtual_machine.BaseVirtualMachine. something along the lines of:  
_result = super(OpenStackVirtualMachine, self).GetResourceMetadata()_
_result[use_provisioning_script] = true_
Any custom flags that was used should be recorded with this method.

File: perfkitbenchmarker/providers/openstack/os_virtual_machine.py [hunk 1]
```diff
@@ -391,6 +394,17 @@ def CreateScratchDisk(self, disk_spec):
...
     self._CreateScratchDiskFromDisks(disk_spec, disks)
 
+  def GetResourceMetadata(self):
+    """Returns a dict containing metadata about the VM.
+
+    Returns:
+      dict mapping string property key to value.
+    """
+    result = super(OpenStackVirtualMachine, self).GetResourceMetadata()
+    if FLAGS.openstack_post_provisioning_script:
```
Comment (jellyfishcake) [inline perfkitbenchmarker/providers/openstack/os_virtual_machine.py]: flag values should not be reused. Store the value as a object attribute, i.e. self.post_provisioning_script = FLAGS.openstack_post_provisioning_script during __init__ and then use self.post_provisioning_script in later calls.

Comment (keerockl) [reply perfkitbenchmarker/providers/openstack/os_virtual_machine.py]: Thank you @jellyfishcake, I have updated patchset.

**Pull request merged**